Remotion/Web/Development.WebTesting/WebDriver/Configuration/Chrome/ChromeSecurityWarningsRegistryCleanUpStrategyFactory.cs

```diff
@@ -0,0 +1,128 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// The re-motion Core Framework is free software; you can redistribute it 
+// and/or modify it under the terms of the GNU Lesser General Public License 
+// as published by the Free Software Foundation; either version 2.1 of the 
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful, 
+// but WITHOUT ANY WARRANTY; without even the implied warranty of 
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// 
+using System;
+using JetBrains.Annotations;
+using Microsoft.Win32;
+
+namespace Remotion.Web.Development.WebTesting.WebDriver.Configuration.Chrome
+{
+  /// <summary>
+  /// Responsible for interacting with the Registry (reading or manipulating the <c>CommandLineFlagSecurityWarningsEnabled</c> entry for Chrome).
+  /// </summary>
+  public class ChromeSecurityWarningsRegistryCleanUpStrategyFactory
+  {
+    private class RestoreSecurityWarningsRegistryCleanUpStrategy : IBrowserSessionCleanUpStrategy
+    {
+      public void CleanUp ()
+      {
+        EnsureWriteAccessToChromePolicyKey();
+
+        using (var key = GetOrCreateChromePoliciesKey())
+        {
+          key.DeleteValue (c_securityWarningsPolicyName);
+        }
+      }
+    }
+
+    private class EmptyCleanUpStrategy : IBrowserSessionCleanUpStrategy
+    {
+      public void CleanUp ()
+      {
+      }
+    }
+
+    private const string c_chromePoliciesPath = @"SOFTWARE\Policies\Google\Chrome";
+    private const string c_securityWarningsPolicyName = "CommandLineFlagSecurityWarningsEnabled";
+
+    private readonly ChromeDisableSecurityWarningsBehavior _disableSecurityWarningsBehavior;
+
+    public ChromeSecurityWarningsRegistryCleanUpStrategyFactory (ChromeDisableSecurityWarningsBehavior disableSecurityWarningsBehavior)
+    {
+      _disableSecurityWarningsBehavior = disableSecurityWarningsBehavior;
+    }
+
+    /// <summary>
+    /// Depending on the set behavior, either throws an exception if <c>CommandLineFlagSecurityWarningsEnabled</c> is not set to false in the registry, attempts to set it,
+    /// or does nothing.
+    /// </summary>
+    /// <returns>
+    /// A <see cref="IBrowserSessionCleanUpStrategy"/>, cleaning up the registry flag if set.
+    /// </returns>
+    /// <exception cref="InvalidOperationException">
+    /// If the behavior <see cref="ChromeDisableSecurityWarningsBehavior.Require"/> is set, and the needed registry key is not set.
+    /// </exception>
+    [NotNull]
+    public IBrowserSessionCleanUpStrategy CreateCleanUpStrategy ()
+    {
+      switch (_disableSecurityWarningsBehavior)
```

Comment:
would it be easier to read if we added a check IF (!Warnings) return new EmptyStrategy() and then followiwth the switch?